src/components/tab-group/tab-group.test.ts

```diff
@@ -0,0 +1,450 @@
+import { aTimeout, elementUpdated, expect, fixture, oneEvent, waitUntil } from '@open-wc/testing';
+import { clickOnElement } from '../../internal/test';
+import { html } from 'lit';
+import { isElementVisibleFromOverflow } from '../../internal/test/element-visible-overflow';
+import { queryByTestId } from '../../internal/test/data-testid-helpers';
+import { sendKeys } from '@web/test-runner-commands';
+import { waitForScrollingToEnd } from '../../internal/test/wait-for-scrolling';
+import type { HTMLTemplateResult } from 'lit';
+import type SlTab from '../tab/tab';
+import type SlTabGroup from './tab-group';
+import type SlTabPanel from '../tab-panel/tab-panel';
+
+interface ClientRectangles {
+  body?: DOMRect;
+  navigation?: DOMRect;
+}
+
+interface CustomEventPayload {
+  name: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const waitForScrollButtonsToBeRendered = (tabGroup: SlTabGroup): Promise<any> => {
+  return waitUntil(() => {
+    const scrollButtons = tabGroup.shadowRoot?.querySelectorAll('sl-icon-button');
+    return scrollButtons?.length === 2;
+  });
+};
```

Comment:
It doesn't look like we use the promise's return value. Should we drop the return and use `Promise<void>` to avoid the eslint rule?

Reply:
No, that is a mistake. I fixed it.

> Thanks again for all your work on this.

No worries, I learned a lot when doing this. Thank you for the helpful comments.